src/nni_manager/core/nniDataStore.ts

```diff
@@ -358,35 +339,94 @@ class NNIDataStore implements DataStore {
+                            id: trialId,
+                            status: this.getJobStatusByLatestEvent('RUNNING', record.event),
+                            hyperParameters: [JSON.stringify(hyperParameter)],
+                            startTime: record.timestamp,
+                            sequenceId: sequenceId++
+                        };
+                        trialMap.set(trialId, trialInfo);
                     }
+                    if (!trialInfo) {
+                        throw new Error('Empty trialInfo');
+                    }
+
+                    let trialIds = jobTrialMap.get(jobInfo.id);
+                    if (trialIds === undefined) {
+                        trialIds = [];
+                        jobTrialMap.set(jobInfo.id, trialIds);
+                    }
+                    trialIds.push(trialId);
                 }
             }
-            if (record.sequenceId !== undefined && jobInfo.sequenceId === undefined) {
-                jobInfo.sequenceId = record.sequenceId;
+        }
+
+        for (const jobTrial of jobTrialMap) {
+            const jobInfo = jobMap.get(jobTrial[0]);
+            const trialIds = jobTrial[1];
+            let trialInfo: TrialJobInfo | undefined;
+            let lastTrialInfo: TrialJobInfo | undefined;
+
+            if (!jobInfo) {
+                throw new Error('Empty jobInfo');
+            }
+
+            for (const trialId of trialIds) {
+                trialInfo = trialMap.get(trialId);
+                if (!trialInfo) {
+                    throw new Error('Empty trialInfo');
+                }
+                if (lastTrialInfo) {
+                    lastTrialInfo.endTime = trialInfo.startTime;
+                }
+                trialInfo.logPath = jobInfo.logPath;
+                // There is no status reported for each trial. So assume it's success, if it's not last one.
+                // The last trial status is the same as job.
+                trialInfo.status = 'SUCCEEDED';
+
+                lastTrialInfo = trialInfo;
+            }
+            // the last one should follow job information
+            if (trialInfo) {
+                trialInfo.endTime = jobInfo.endTime;
+                trialInfo.status = jobInfo.status;
             }
-            map.set(record.trialJobId, jobInfo);
         }
+        this.log.debug('getTrialsByReplayEvents done');
 
-        this.log.debug('getTrialJobsByReplayEvents done');
+        return trialMap;
+    }
 
-        return map;
+    private getTrialId(jobId?: string, metricData?: MetricData): string {
+        let trialId = "";
+        let parameterId = "";
+        if (metricData) {
+            if (metricData.trial_job_id) {
+                trialId = metricData.trial_job_id;
+            } else if (jobId) {
+                trialId = jobId;
+            } else {
+                throw new NNIError("argument null!", "jobId and metricData.trial_job_id shouldn't be undefined both!");
+            }
```

Comment:
suggest to have a simpler rule to generate trial id, such as `jobId-parameterId`

Reply:
updated